cmd/podman/images/scp.go

```diff
@@ -0,0 +1,329 @@
+		ValidArgsFunction: common.AutocompleteImages,
+		Example:           `podman image scp myimage:latest otherhost::`,
+	}
+)
+
+var (
+	scpOpts entities.ImageScpOptions
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: imageScpCommand,
+		Parent:  imageCmd,
+	})
+	scpFlags(imageScpCommand)
+}
+
+func scpFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.BoolVarP(&scpOpts.Save.Quiet, "quiet", "q", false, "Suppress the output")
+}
+
+func scp(cmd *cobra.Command, args []string) (finalErr error) {
+	var (
+		//TODO add tag support for images
+		err error
+	)
+	if scpOpts.Save.Quiet { // set quiet for both load and save
+		scpOpts.Load.Quiet = true
+	}
+	f, err := ioutil.TempFile("", "podman") // open temp file for load/save output
+	if err != nil {
+		return err
+	}
+	defer os.Remove(f.Name())
+
+	scpOpts.Save.Output = f.Name()
+	scpOpts.Load.Input = scpOpts.Save.Output
+	if err := parse.ValidateFileName(saveOpts.Output); err != nil {
+		return err
+	}
+	confR, err := config.NewConfig("") // create a hand made config for the remote engine since we might use remote and native at once
+	if err != nil {
+		return errors.Wrapf(err, "could not make config")
+	}
+	abiEng, err := registry.NewImageEngine(cmd, args) // abi native engine
+	if err != nil {
+		return err
+	}
+
+	cfg, err := config.ReadCustomConfig() // get ready to set ssh destination if necessary
+	if err != nil {
+		return err
+	}
+	urls, iden, serv, err := parseArgs(args, cfg) // parses connection data and "which way" we are loading and saving
+	if err != nil {
+		return err
+	}
+	// TODO: Add podman remote support
+
+	uri := []*urlP.URL{}
+	for i, url := range urls {
+		urlt, err := urlP.Parse(url) // create an actual url to pass to exec command
+		if err != nil {
+			return err
+		}
+		uri = append(uri, urlt)
+		if uri[i].User.Username() == "" {
+			if uri[i].User, err = connection.GetUserInfo(uri[i]); err != nil {
+				return err
+			}
+		}
+	}
+
+	confR.Engine = config.EngineConfig{Remote: true, CgroupManager: "cgroupfs", ServiceDestinations: serv} // pass the service dest (either remote or something else) to engine
+
+	// TODO: add podman remote support
+
+	switch {
+	case scpOpts.FromRemote: // if we want to load FROM the remote
+		_, err := execCommand(iden[0], uri[0], scpOpts.Save.Output, scpOpts.DestImageName, "", false)
+		if err != nil {
+			return err
+		}
+		if scpOpts.ToRemote { // we want to load remote -> remote
+			rep, err := execCommand(iden[1], uri[1], scpOpts.Save.Output, scpOpts.Load.Input, scpOpts.Tag, true) // validate connection given and execute save or load on remote machine
+			if err != nil {
+				return err
+			}
+			fmt.Println(rep)
+			break
+		}
+		report, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(report.Names, ","))
+	case scpOpts.ToRemote: // remote host load
+		scpOpts.Save.Format = "oci-archive"
```

Comment:
Why is this used only for this case? I’d expect the best format to be (possibly image-dependent but) the same for all copy directions.

Reply:
added it in the remote save so now it is equal. load and save only take particular formats used together.